tests/unit/parallel/algorithms/foreach_executors.cpp

```diff
@@ -0,0 +1,92 @@
+//  Copyright (c) 2015 Daniel Bourgeois
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/hpx_init.hpp>
+#include <hpx/hpx.hpp>
+
+#include "foreach_tests.hpp"
+
+////////////////////////////////////////////////////////////////////////////////
+template <typename ExPolicy>
+void test_executors(ExPolicy && policy)
+{
+    using iterator_tag = std::random_access_iterator_tag;
+
+    test_for_each(std::forward<ExPolicy>(policy), iterator_tag());
+    test_for_each_exception(std::forward<ExPolicy>(policy), iterator_tag());
+    test_for_each_bad_alloc(std::forward<ExPolicy>(policy), iterator_tag());
```

Comment:
While it may work, I don't think policy should be forwarded multiple times. And it is redundant since test_for_each accepts a universal reference for testing purposes but the other two don't. Will change.